cli/internal/cmd/miniup.go

```diff
@@ -0,0 +1,268 @@
+	cmd.Printf("\tvirsh -c %s\n\n", connectURI)
+
+	// initialize cluster
+	if err := initializeMiniCluster(cmd, fileHandler); err != nil {
+		return fmt.Errorf("initializing cluster: %w", err)
+	}
+	return nil
+}
+
+// checkSystemRequirements checks if the system meets the requirements for running a mini Constellation cluster.
+// We do so by verifying that the host:
+// - arch/os is linux/amd64.
+// - has access to /dev/kvm.
+// - has more than 4 CPU cores.
+// - has more than 4GB of memory.
+func checkSystemRequirements(out io.Writer) error {
+	// check arch/os
+	if runtime.GOARCH != "amd64" || runtime.GOOS != "linux" {
+		return fmt.Errorf("creation of a QEMU based Constellation is not supported for %s/%s", runtime.GOOS, runtime.GOARCH)
+	}
+
+	// check if /dev/kvm exists
+	if _, err := os.Stat("/dev/kvm"); err != nil {
+		return fmt.Errorf("unable to access KVM device: %w", err)
+	}
+
+	// check CPU cores
+	if runtime.NumCPU() < 4 {
+		return fmt.Errorf("insufficient CPU cores: %d, at least 4 cores are required by a QEMU cluster", runtime.NumCPU())
+	}
+	if runtime.NumCPU() < 6 {
+		fmt.Fprintf(out, "WARNING: Only %d CPU cores available. This may cause performance issues.\n", runtime.NumCPU())
+	}
+
+	// check memory
+	f, err := os.Open("/proc/meminfo")
+	if err != nil {
+		return fmt.Errorf("determining available memory: failed to open /proc/meminfo: %w", err)
+	}
+	defer f.Close()
+	var memTotal int
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		if strings.HasPrefix(scanner.Text(), "MemTotal:") {
+			_, err = fmt.Sscanf(scanner.Text(), "MemTotal:%d", &memTotal)
+			if err != nil {
+				return fmt.Errorf("determining available memory: failed to parse /proc/meminfo: %w", err)
+			}
+		}
+	}
+	memGB := memTotal / 1024 / 1024
+	if memGB < 4 {
+		return fmt.Errorf("insufficient memory: %dGB, at least 4GB of memory are required by a QEMU cluster", memTotal)
+	}
+	if memGB < 6 {
+		fmt.Fprintln(out, "WARNING: Less than 6GB of memory available. This may cause performance issues.")
+	}
+
+	return nil
+}
+
+// prepareConfig reads a given config, or creates a new minimal QEMU config.
+func prepareConfig(cmd *cobra.Command, fileHandler file.Handler) (*config.Config, error) {
+	configPath, err := cmd.Flags().GetString("config")
+	if err != nil {
+		return nil, err
+	}
+
+	// check for existing config
+	if configPath != "" {
+		config, err := readConfig(cmd.OutOrStdout(), fileHandler, configPath)
+		if err != nil {
+			return nil, err
+		}
+		if config.GetProvider() != cloudprovider.QEMU {
+			return nil, errors.New("invalid provider for mini constellation cluster")
+		}
+		return config, nil
+	}
+
+	// download image to current directory if it doesn't exist
+	imagePath := "./constellation.qcow2"
+	if _, err := os.Stat(imagePath); err == nil {
+		cmd.Printf("Using existing image at %s\n\n", imagePath)
+	} else if errors.Is(err, os.ErrNotExist) {
+		cmd.Println("Downloading image to ./constellation.qcow2")
```

Comment:
```suggestion
		cmd.Println("Downloading image to", imagePath)
```

It's practice it's the same output, but still in case this is edited in the future / becomes user-configurable...

Reply:
Leftover code from when it was user-configurable (It should no longer be)